app/login/index.js

```diff
@@ -31,12 +33,23 @@ exports.loginService = function loginService(parentWindow, callback) {
 	win.loadURL(`file://${__dirname}/login.html`);
 };
 
-exports.handleLoginDialogTry = function handleLoginDialogTry(window) {
+exports.handleLoginDialogTry = function handleLoginDialogTry(window, {ssoUser, ssoPasswordCommand}) {
+	console.log(ssoUser, ssoPasswordCommand);
 	window.webContents.on('login', (event, request, authInfo, callback) => {
 		event.preventDefault();
 		if (isFirstLoginTry) {
 			isFirstLoginTry = false;
-			this.loginService(window, callback);
+			if (ssoUser && ssoPasswordCommand) {
+				//TODO: Implement a timeout ?
+				console.log(`Retrieve password using command : ${ssoPasswordCommand}`);
+				exec(ssoPasswordCommand, (error, stdout, stderr) => {
```

Comment:
Fairly new to nodejs, maybe document if this is running in a shell or not (to be able to use pipes ect in the provided `ssoPasswordCommand`

Reply:
You can simplify this by using execSync. Something like:

```suggestion
 try {
    const ssoPassword = execSync(ssoPasswordCommand).toString();
    callback(ssoUser, ssoPassword);
 } catch (error) {
    console.error(`Failed to execute ssoPasswordCommand. Status Code: ${error.status} with '${error.message}'`);
 }
```

I haven't tried the code but that should execute it synchronously and I think that is ok in this case.